sdk/formrecognizer/ai-form-recognizer/MIGRATION-v3_v4.md

```diff
@@ -0,0 +1,232 @@
+
+## Partial Migration (Side-by-Side)
+
+To avoid migrating an application all at once, major version 3 may be installed alongside major version 4 using a dependency alias. Either version may be aliased. For example, to install the new SDK package under an alias, add the following to the `dependencies` field of `package.json`:
+
+```javascript
+{
+    ...,
+    "dependencies": {
+        ...,
+        "@azure/ai-form-recognizer": "^3.2.0",
+        "@azure/ai-form-recognizer-beta": "npm:@azure/ai-form-recognizer@4.0.0-beta.1"
+    }
+}
+```
+
+With this configuration, imports from `"@azure/ai-form-recognizer"` will import from major version 3 of the package, and imports from `"@azure/ai-form-recognizer-beta"` will import from the new beta package. Of course, major version 3 could be aliased instead:
+
+```javascript
+{
+    ...,
+    "dependencies": {
+        ...,
+        "@azure/ai-form-recognizer": "4.0.0-beta.1",
+        "@azure/ai-form-recognizer-v3": "npm:@azure/ai-form-recognizer@^3.2.0"
+    }
+}
+```
+
+Then, the two packages may be used side-by-side, and an application may be migrated partially or over time:
+
+```javascript
+import { FormTrainingClient } from "@azure/ai-form-recognizer-v3";
+import { DocumentAnalysisClient } from "@azure/ai-form-recognizer";
+```
+
+## Understanding the New Return Types
+
+In the new 4.0.0 SDK package, several types have been introduced, renamed, restructured, and removed. This section describes the differences between the packages.
+
+### Terminology
+
+Some terminology has changed to reflect the enhanced capabilities of the newest Form Recognizer service APIs. While the service is still called "Form Recognizer," it is capable of much more than simple recognition, and is not limited to documents that are "forms". As a result, we've made the following broad changes to the terminology used throughout the SDK:
+
+- The word "Document" has broadly replaced the word "Form." The service supports a wide variety of documents and data-extraction scenarios, not merely limited to "forms."
+- The word "Analyze" has broadly replaced the word "Recognize." The document analysis operation executes a data extraction pipeline that supports more than just recognition.
+- Distinctions between "custom" and "prebuilt" models have broadly been eliminated. Prebuilt models are simply models that were created by the Form Recognizer service team and that exist within every Form Recognizer resource.
+- The concept of "model training" has broadly been replaced with "model creation" or "model administration" (whatever is most appropriate in context), as not all model creation operations involve "training" a model from a data set. When referring to a model schema trained from a data set, we will use the term "document type" instead.
```

Comment:
I like this section. what do you guys think about adding it the other migration guides? @catalinaperalta @maririos @samvaity

Reply:
I like it, I did consider mentioning this in our migration guide but wasn't too sure how much to explain. I like this description, I'll pass this over to ours.